### ISSUE TITLE
High contrast issues in components

### ISSUE THREAD

**Issue by betrue-final-final** [OfficeDev/office-ui-fabric-react#3102]
Are you willing to submit a PR to fix? (Yes)

Requested priority: (Normal) 

Hey, I was looking through Master in high contrast and noticed some issues:

1. Carets in the UI are still grey, and should be windowText system color
![image](https://user-images.githubusercontent.com/20363037/31502736-c44d31ac-af22-11e7-8348-0a2348d5cc7c.png)

2. Cards should have a windowText color border around them
![image](https://user-images.githubusercontent.com/20363037/31502798-eb319d76-af22-11e7-828a-86f2317f0908.png)

3. ChoiceGroup with icons has grey icons for disabled. Should use grayText system color
![image](https://user-images.githubusercontent.com/20363037/31502881-112603aa-af23-11e7-9e22-9a72af7c5636.png)

4. Borders around people in people picker make it hard to tell where focus is
![image](https://user-images.githubusercontent.com/20363037/31502915-23f399d4-af23-11e7-9490-22a9b9dad6f8.png)
Same for SwatchColorPicker
![image](https://user-images.githubusercontent.com/20363037/31502960-3e70c5d4-af23-11e7-81dd-1c0683e99873.png)

5. Disabled dropdown does not have a border
![image](https://user-images.githubusercontent.com/20363037/31502991-530b5a9a-af23-11e7-9c51-95d0874cd770.png)

6. Personas have an extraneous square around them.
![image](https://user-images.githubusercontent.com/20363037/31503029-66e805a4-af23-11e7-91dc-f1517a05a4c5.png)

7. Theme Generator only shows grey
![image](https://user-images.githubusercontent.com/20363037/31503092-93bbac52-af23-11e7-831b-45ff665e7fb0.png)

8. We should consider reversing primary buttons so people can distinguish them
![image](https://user-images.githubusercontent.com/20363037/31503167-cef95a58-af23-11e7-95d9-e2f4a22535e1.png)

9. Why is upload link color and the others aren't?
![image](https://user-images.githubusercontent.com/20363037/31503197-e5c2c85a-af23-11e7-8463-1ebb56c69fce.png)

10. MessageBar close X has weird square brackets around it
![image](https://user-images.githubusercontent.com/20363037/31503231-f9561872-af23-11e7-9bf6-c8db279f715a.png)

11. Detailslist hover selection check should have a circle border
![image](https://user-images.githubusercontent.com/20363037/31503267-0b832918-af24-11e7-9997-b58543a52456.png)

12. We may consider making focus 2px for Detailslist, as it's hard to see it
![image](https://user-images.githubusercontent.com/20363037/31503297-1c0fefbe-af24-11e7-93a3-84fd35a05f74.png)

13. There's a black dot in the middle of toggle thumbs
![image](https://user-images.githubusercontent.com/20363037/31503350-3fa78cde-af24-11e7-84a9-c90eb4047651.png)

14. We should reverse the "on" toggle for better visibility
![image](https://user-images.githubusercontent.com/20363037/31503426-7d56c5b8-af24-11e7-9ed3-3f7e8d3bc13b.png)

15. Progress bar is hard to see. We may consider dropping the top line of the empty track so the fill an be seen better. 
![image](https://user-images.githubusercontent.com/20363037/31503707-398ff088-af25-11e7-9c6a-f84875ddc648.png)

16. Why does spinner look so weird in high contrast?
![image](https://user-images.githubusercontent.com/20363037/31503800-73bec126-af25-11e7-905a-c3bca9cc3ecd.png)

**Comment by kristoferbrown** [comment]
@betrue-final-final, a couple questions as I'm starting to go down this list: 
- What component does 1 occur in? I can't see that happen anywhere.
- Can you still repro 2 in master? It appears to have been fixed by the last HC bugfix PR.
- Item 6 is occurring due to a regression preventing the WindowText border I put around initial only personas from being a circle. This got me thinking, is there a better way to handle initial only personas, or is just a border circle enough?

**Comment by betrue-final-final** [comment]
#1 seems to be a no-repro now
#2 is no repro!
For high contrast, we don't have backgrounds, just borders. Alternately we could reverse them and make them white with black initials.

**Comment by Hardhik** [comment]
Please add this to the above list.
https://developer.microsoft.com/en-us/fabric#/components/dropdown
The focus on dropdown is not visible in high contrast white mode.

![image](https://user-images.githubusercontent.com/2673006/32712800-76b9c8ca-c86c-11e7-8154-a4b4c88de2bb.png)

**Comment by kristoferbrown** [comment]
@Hardhik, interesting it looks like the dark purple color that the default black-on-white high contrast palette uses for selected items is close enough to to black that it's really hard to see the border change colors when focused. 

@betrue-final-final, do you think this is enough of an issue that we should rethink the focus styling on this component? The fix would probably be to make the background of the component (or just a highlight under the text) be the `Highlight` color instead of changing the border color. What do you think?
